prov/shm/src/smr_ep.c

```diff
@@ -1621,7 +1239,9 @@ int smr_srx_context(struct fid_domain *domain, struct fi_rx_attr *attr,
 		smr_domain->srx = ((struct fi_peer_srx_context *) (context))->srx;
 		return FI_SUCCESS;
 	}
-	return smr_ep_srx_context(smr_domain, attr->size, rx_ep);
+	return util_ep_srx_context(&smr_domain->util_domain, attr->size,
+			SMR_IOV_LIMIT, SMR_INJECT_SIZE, &smr_update,
+			&smr_domain->util_domain.lock, rx_ep);
```

Comment:
I don't think we need to check the threading level on the domain lock. The FI_THREAD_DOMAIN specification that we optimize for in the EP specifies that:
`A domain serialization model requires applications to serialize access to all objects belonging to a domain.`
So that means that anything within the domain is serialized, but not necessarily the domain itself.